nsqd/protocol_v2.go

```diff
@@ -615,7 +615,10 @@ func (p *protocolV2) SUB(client *clientV2, params [][]byte) ([]byte, error) {
 	for {
 		topic := p.ctx.nsqd.GetTopic(topicName)
 		channel = topic.GetChannel(channelName)
-		channel.AddClient(client.ID, client)
+		if err := channel.AddClient(client.ID, client); err != nil {
+			return nil, protocol.NewFatalClientErr(nil, "E_TOO_MANY_CHANNEL_CONSUMERS",
+				fmt.Sprintf("TOO many channel consumers for %s:%s", topicName, channelName))
```

Comment:
Last nit, let's change this message to be:

```go
fmt.Sprintf("channel consumers for %s:%s exceeds limit of %d", topicName, channelName, ...)
```

🙏

Reply:
Will do.

Reply:
@mreiferson Done. PTAL again. :)